File: 2023/advent_of_code/test/q5/q5_test.exs
defmodule AdventOfCode.Q5Test do
  use ExUnit.Case
  doctest AdventOfCode.Q5

  test "Q5 Part 1 Example works" do
    file_stream = File.stream!("./test/q5/q5_example.txt")
    result = AdventOfCode.Q5.part1(file_stream)
    assert result == 35
  end

  test "Q5 Part 1 works" do
    file_stream = File.stream!("./test/q5/q5.txt")
    result = AdventOfCode.Q5.part1(file_stream)
    assert result == 484_023_871
  end

  test "Q5 Part 2 Example works" do
    file_stream = File.stream!("./test/q5/q5_example.txt")
    result = AdventOfCode.Q5.part2(file_stream)
    assert result == 46
  end

  test "Q5 Part 2 works" do
    file_stream = File.stream!("./test/q5/q5.txt")
    result = AdventOfCode.Q5.part2(file_stream)
    assert result == 46_294_175
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q5/Q5Test.kt
package q5

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q5Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q5.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals("VQZNJMWTR", part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q5.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals("NLCDCLVMQ", part2(filePath!!))
    }
}